exporter/exporter.go

```diff
@@ -0,0 +1,160 @@
+	// Optional.
+	MonitoringClientOptions option.ClientOption
+
+	// TraceClientOptions are additional options to be passed
+	// to the underlying Stackdriver Trace API client.
+	// Optional.
+	TraceClientOptions []option.ClientOption
+
+	// TraceSpansBufferMaxBytes is the maximum size (in bytes) of spans that
+	// will be buffered in memory before being dropped.
+	//
+	// If unset, a default of 8MB will be used.
+	// TraceSpansBufferMaxBytes int
+
+	// DefaultTraceAttributes will be appended to every span that is exported to
+	// Stackdriver Trace.
+	DefaultTraceAttributes map[string]interface{}
+
+	// Context allows you to provide a custom context for API calls.
+	//
+	// This context will be used several times: first, to create Stackdriver
+	// trace and metric clients, and then every time a new batch of traces or
+	// stats needs to be uploaded.
+	//
+	// Do not set a timeout on this context. Instead, set the Timeout option.
+	//
+	// If unset, context.Background() will be used.
+	Context context.Context
+
+	// SkipCMD enforces to skip all the CreateMetricDescriptor calls.
+	// These calls are important in order to configure the unit of the metrics,
+	// but in some cases all the exported metrics are builtin (unit is configured)
+	// or the unit is not important.
+	SkipCMD bool
+
+	// Timeout for all API calls. If not set, defaults to 5 seconds.
+	Timeout time.Duration
+
+	// ReportingInterval sets the interval between reporting metrics.
+	// If it is set to zero then default value is used.
+	ReportingInterval time.Duration
+
+	// NumberOfWorkers sets the number of go rountines that send requests
+	// to Stackdriver Monitoring. This is only used for Proto metrics export
+	// for now. The minimum number of workers is 1.
+	NumberOfWorkers int
+
+	// MetricDescriptorTypeFormatter is the custom formtter for the MetricDescriptor.Type.
+	// By default, the format string is "custom.googleapis.com/opentelemetry/[metric name]".
+	MetricDescriptorTypeFormatter func(*apimetric.Descriptor) string
+}
+
+
+// Exporter is a trace and metric exporter
+type Exporter struct {
+	traceExporter *cloudtrace.Exporter
+}
+
+
+// NewExporter creates a new Exporter that implements both trace.Exporter
+// and metric.Exporter
+func NewExporter(o Options) (*Exporter, error) {
+
+	te, err := cloudtrace.NewExporter(cloudtrace.WithProjectID(o.ProjectID), cloudtrace.WithContext(o.Context),
+		cloudtrace.WithTraceClientOptions(o.TraceClientOptions), cloudtrace.WithTimeout(o.Timeout), 
+		cloudtrace.WithOnError(o.OnError),
+	)
+	if err != nil {
+		return nil, err
+	} 
+	
+	return &Exporter{
+		traceExporter: te,
+	}, nil
+}
+
+// readMonitoredResourcesFromMetricsExporter obtains Monitored Resources labels from metrics exporter
+func readMonitoredResourcesFromMetricsExporter() (resType string, labels map[string]string) {
+	//TODO: read the resources from metric. Now the autodetect is for mockup
+	return monitoredresource.Autodetect().MonitoredResource()
```

Comment:
AFAIK the long term plan was to move auto-detection to https://github.com/open-telemetry/opentelemetry-go, so that it can be reused by any exporter. 
The only problem I am aware of is that OpenTelemetry resources don't have the dedicated "type" field, so we have to define it as a "well known" resource label (convention).

For the short term implementing auto-detection in the exporter might be fine, but we shouldn't use OpenCensus code here, I think. So maybe we should copy the code over to this repo, and eventually extract it to `opentelemetry-go`?

@james-bebbington may have other opinion / know more details.